dspace-api/src/main/java/org/dspace/subscriptions/ContentGenerator.java

```diff
@@ -0,0 +1,92 @@
+
+import java.io.ByteArrayOutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Resource;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.Item;
+import org.dspace.content.crosswalk.StreamDisseminationCrosswalk;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.core.Email;
+import org.dspace.core.I18nUtil;
+import org.dspace.discovery.IndexableObject;
+import org.dspace.eperson.EPerson;
+import org.dspace.subscriptions.service.SubscriptionGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation class of SubscriptionGenerator
+ * which will handle the logic of sending the emails
+ * in case of 'content' subscriptionType
+ */
+@SuppressWarnings("rawtypes")
+public class ContentGenerator implements SubscriptionGenerator<IndexableObject> {
+
+    private final Logger log = LogManager.getLogger(ContentGenerator.class);
+
+    @SuppressWarnings("unchecked")
+    @Resource(name = "entityDissemination")
+    private Map<String, StreamDisseminationCrosswalk> mapEntityDisseminatorProperty = new HashMap();
+
+    @Autowired
+    private ItemService itemService;
+
+    @Override
+    public void notifyForSubscriptions(Context context, EPerson ePerson,
+                                       List<IndexableObject> indexableComm,
+                                       List<IndexableObject> indexableColl) {
+        try {
+            if (Objects.nonNull(ePerson)) {
+                Locale supportedLocale = I18nUtil.getEPersonLocale(ePerson);
+                Email email = Email.getEmail(I18nUtil.getEmailFilename(supportedLocale, "subscriptions_content"));
+                email.addRecipient(ePerson.getEmail());
+                email.addArgument(generateBodyMail(context, indexableComm));
+                email.addArgument(generateBodyMail(context, indexableColl));
+                email.send();
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+            log.warn("Cannot email user eperson_id: {} eperson_email: {}", ePerson::getID, ePerson::getEmail);
+        }
+    }
+
+    private String generateBodyMail(Context context, List<IndexableObject> indexableObjects) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            out.write("\n".getBytes(UTF_8));
+            if (indexableObjects.size() > 0) {
+                for (IndexableObject indexableObject : indexableObjects) {
+                    out.write("\n".getBytes(UTF_8));
+                    Item item = (Item) indexableObject.getIndexedObject();
+                    String entityType = itemService.getEntityTypeLabel(item);
+                    mapEntityDisseminatorProperty.get(entityType).disseminate(context, item, out);
```

Comment:
I'm getting a NullPointerException on this line whenever I'm supposed to receive an email related to a normal Item object (non-Entity).  It appears this code doesn't work for default Items... it requires you to have Entities.  Here's the NullPointer I see when I run `./dspace subscription-send -f W`:

```
2023-02-07 15:26:26,922 ERROR unknown unknown org.dspace.subscriptions.ContentGenerator @ null
java.lang.NullPointerException: null
        at org.dspace.subscriptions.ContentGenerator.generateBodyMail(ContentGenerator.java:79) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.subscriptions.ContentGenerator.notifyForSubscriptions(ContentGenerator.java:60) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.subscriptions.SubscriptionEmailNotificationServiceImpl.perform(SubscriptionEmailNotificationServiceImpl.java:101) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.subscriptions.SubscriptionEmailNotification.internalRun(SubscriptionEmailNotification.java:60) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.scripts.DSpaceRunnable.run(DSpaceRunnable.java:104) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.app.launcher.ScriptLauncher.executeScript(ScriptLauncher.java:149) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.app.launcher.ScriptLauncher.handleScript(ScriptLauncher.java:131) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
        at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:98) [dspace-api-7.5-SNAPSHOT.jar:7.5-SNAPSHOT]
```